python/dialects/aiex.py

```diff
@@ -836,3 +836,64 @@ def dma_start_bd_chain_for(symbol, args, alloc, *pyargs, **kwargs):
     return DMAStartBdChainForOp(
         T.index(), chain_sym, args, alloc_sym, *pyargs, **kwargs
     )
+
+
+def shim_dma_bd(
+    mem,
+    offset: int | None = None,
+    sizes: MixedValues | None = None,
+    strides: MixedValues | None = None,
+    transfer_len: int | None = None,
+):
+    if offset is None:
+        offset = 0
+    if sizes is None:
+        sizes = [0] * 4
+    if strides is None:
+        strides = [0] * 3 + [1]
+
+    if transfer_len is None:
+        if len(sizes) >= 4:
+            # For shim dma bd, highest dimension is repeat count which is not included in the length
+            transfer_len = np.prod(sizes[1:])
+        else:
+            # If does not have highest dimension, then we can take the product of all dimensions
+            transfer_len = np.prod(sizes)
```

Comment:
In Python, we generally don't enforce having exactly 4 dimensions. Unless you want to start checking number of dimensions and outputting a useful error message in Python (duplicated from the checks in C++ code) then we need to avoid unchecked assumptions as to the length of the sizes/strides lists/

Reply:
The code I suggested should work with any number of dimensions, it does not require 4.

Reply:
That being said your comment maybe proves that it is not, in fact, clearer. haha!

Reply:
Ack! You are right, whoops! I will change this. Thanks!